backend/internal/service/handler/reviews/vote.go

```diff
@@ -0,0 +1,40 @@
+package reviews
+
+import (
+	"github.com/gofiber/fiber/v2"
+
+	"platnm/internal/errs"
+)
+
+type createVoteRequest struct {
+	models.UserReviewVote
+}
+
+func (h *Handler) VoteReview(c *fiber.Ctx) error {
+	var req createVoteRequest
+	if err := c.BodyParser(&req); err != nil {
+		return errs.BadRequest("failed to parse request body")
+	}
+
+	exists, err := h.userRepository.UserExists(c.Context(), &req.UserReviewVote.UserID)
+	if err != nil {
+		return err
+	}
+
+	if !exists {
+		return errs.NotFound("User", "id", &req.UserReviewVote.UserID)
+	}
+	exists, err := h.reviewRepository.ReviewExists(c.Context(),&req.UserReviewVote.ReviewID)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return errs.NotFound("Review", "id", &req.UserReviewVote.ReviewID)
+	}
+	vote, err := h.voteRepository.AddVote(c.Context(), &req.UserReviewVote)

```

Comment:
before we go ahead and add a vote, we probably want a VoteExists function to see if the user has already reacted to the provided review. this leads us to 3 cases:
1. Vote does not exist: just add it
2. Vote exists with the same "upvote" value (ex. you hit upvote after you've already upvoted a review) -> we should delete the row in the table with a new delete helper method in the VoteRepository
3. Vote exists with the opposite "upvote" value (ex. you previously upvoted a review but now hit downvote) -> we can update the row in the table to have the new upvote value. (two ways: can use a new UpdateVote method in VoteRepository to update the existing row, or you can combine the two existing methods - first delete the old row, then add the new row with the new upvote value). first option is a little more code but a little more efficient

let me know if you have questions!

Reply:
OK I added all of the mentioned functions (went with UpdateVote method). Went in more detail on slack!